app/src/main/java/at/bitfire/icsdroid/db/entity/Subscription.kt

```diff
@@ -0,0 +1,396 @@
+            .updateStatusNotModified(id, lastSync)
+
+    /**
+     * Updates the status of a subscription that has just been modified. This removes its [Subscription.errorMessage], and updates the [Subscription.eTag],
+     * [Subscription.lastModified] and [Subscription.lastSync].
+     * @param context The context that is making the request.
+     * @param eTag The new eTag to set.
+     * @param lastModified The new date to set for [Subscription.lastModified].
+     * @param lastSync The last synchronization date to set. Defaults to the current system time, so can be skipped.
+     * @throws SQLException If any error occurs with the update.
+     */
+    @WorkerThread
+    fun updateStatusSuccess(
+        context: Context,
+        eTag: String? = this.eTag,
+        lastModified: Long? = this.lastModified,
+        lastSync: Long = System.currentTimeMillis()
+    ) = AppDatabase.getInstance(context)
+        .subscriptionsDao()
+        .updateStatusSuccess(id, eTag, lastModified ?: 0L, lastSync)
+
+    /**
+     * Updates the error message of the subscription.
+     * @param context The context that is making the request.
+     * @param message The error message to give to the subscription.
+     * @throws SQLException If any error occurs with the update.
+     */
+    @WorkerThread
+    fun updateStatusError(context: Context, message: String?) =
+        AppDatabase.getInstance(context)
+            .subscriptionsDao()
+            .updateStatusError(id, message)
+
+    /**
+     * Updates the [Subscription.url] field to the given one.
+     * @param context The context that is making the request.
+     * @param url The new url to set.
+     * @throws SQLException If any error occurs with the update.
+     */
+    @WorkerThread
+    fun updateUrl(context: Context, url: Uri) =
+        AppDatabase.getInstance(context)
+            .subscriptionsDao()
+            .update(
+                copy(url = url)
+            )
+
+    /**
+     * Updates the [Subscription.url] field to the given one.
+     * @param context The context that is making the request.
+     * @param url The new url to set.
+     * @throws SQLException If any error occurs with the update.
+     * @throws MalformedURLException If the given [url] cannot be parsed to [Uri].
+     */
+    @WorkerThread
+    fun updateUrl(context: Context, url: String) = updateUrl(context, Uri.parse(url))
+
+    /**
+     * Provides an [AndroidCalendar] from the current subscription.
+     * @param context The context that is making the request.
+     * @return A new calendar that matches the current subscription.
+     * @throws NullPointerException If a provider could not be obtained from the [context].
+     * @throws FileNotFoundException If the calendar is not available in the system's database.
+     */
+    fun getCalendar(context: Context) = AndroidCalendar.findByID(
+        account,
+        getProvider(context)!!,
+        LocalCalendar.Factory,
+        id
+    )
```

Comment:
I wouldn't mix the Subscription with AndroidCalendar. The interface between our subscriptions and the Android calendars should be somewhere else, maybe in `LocalCalendar` or a separate class (whatever results in a better design that separates the components).

Same for all other Android access methods.